Update compute_asset_path to Rails 4 signature

fofr opened the pull request:
Fixes https://github.com/searls/jasmine-rails/issues/213

The JasmineRails::OfflineAssetPaths monkey patch of `compute_asset_path` has a Rails 3 signature:
https://github.com/searls/jasmine-rails/blob/master/lib/jasmine_rails/offline_asset_paths.rb#L9

This was updated from 3 arguments to 2 in Rails 4:
Rails 3: https://apidock.com/rails/ActionView/AssetPaths/compute_public_path
Rails 4: https://apidock.com/rails/ActionView/Helpers/AssetUrlHelper/compute_asset_path

If JasmineRails::OfflineAssetPaths.disabled is true, then the method is meant to shortcut and run the original super method. This gave an error like:

```
ActionView::Template::Error: wrong number of arguments (given 3, expected 1..2)
```

cc @h-lame

fofr commented:
Test failures look unrelated.

searls commented:
We still support Rails 3, though, so could you put this method definition behind a branch that checks version?

fofr commented:
@searls I've updated in 9e8794f to maintain Rails 3 support.

Thought about a switching branch based on Rails version but there was already a Rails 3 entry point on line 30. I've updated this to be explicit about the Rails 3 arguments (they match the old methods before https://github.com/searls/jasmine-rails/pull/204).

For Rails 3:
* If offline assets are disabled it will call `super` and pass in the original arguments, including any `dir` value that's been set
* If enabled it will call `compute_asset_path` and ignore the unused `dir` variable

fofr commented:
@searls Is there anything else I need to update following the changes added in 9e8794f?

searls commented:
Sorry Paul, you're on my list, but I don't want to merge until I fix the build, which is apparently due to a nokogiri upgrade that dropped support for old Rubbies. If you have the time to look into that build failure and fix it in this PR, I will merge it in

> On Aug 22, 2017, at 10:59 AM, Paul Hayes <notifications@github.com> wrote:
> 
> @searls <https://github.com/searls> Is there anything else I need to update following the changes added in 9e8794f <https://github.com/searls/jasmine-rails/commit/9e8794f6e74bfbdbc7b3f71fad98a7725e5b9133>?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/searls/jasmine-rails/pull/214#issuecomment-324053703>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAE1x8o6P2lTMX-5PtwhVFAoO8jIcHA3ks5sauzZgaJpZM4O2oEv>.
>

timdiggins commented:
@fofr I think the build probs now fixed by #219 so rebasing on master should sort it (ref: https://travis-ci.org/timdiggins/jasmine-rails/builds/276747009)

fofr commented:
@timdiggins Thanks Tim, rebased and tests are now passing.

searls commented:
Sorry for the delay in reviewing this. Looks good to me.
